### PR TITLE
4 packages from gitlab.com/nomadic-labs/data-encoding/-/archive/v1.0.1/data-encoding-v1.0.1.tar.gz

### PR DESCRIPTION
**Pull request by raphael-proust**
This pull-request concerns:
-`data-encoding.1.0.1`: Library of JSON and binary encoding combinators
-`json-data-encoding.1.0.1`: Type-safe encoding to and decoding from JSON
-`json-data-encoding-browser.1.0.1`: Type-safe encoding to and decoding from JSON (browser support)
-`json-data-encoding-bson.1.0.1`: Type-safe encoding to and decoding from JSON (bson support)

---

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

**Comment by raphael-proust** [comment]
https://github.com/janestreet/zarith_stubs_js/issues/10 ← this is meant to fix the underlying issue.

**Comment by avsm** [comment]
Still tests failing here...

**Comment by raphael-proust** [comment]
After https://github.com/ocaml/opam-repository/pull/24755 is merged this here MR should be updated with new constraints.

**Comment by raphael-proust** [comment]
Updated with zarith stubs constraints

File: packages/json-data-encoding/json-data-encoding.1.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+maintainer: "contact@nomadic-labs.com"
+authors: [ "Nomadic Labs" "Ocamlpro" ]
+homepage: "https://gitlab.com/nomadic-labs/json-data-encoding"
+bug-reports: "https://gitlab.com/nomadic-labs/json-data-encoding/issues"
+license: "MIT"
+dev-repo: "git+https://gitlab.com/nomadic-labs/json-data-encoding"
+
+build: [
+  ["dune" "build" "-j" jobs "-p" name]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
```
Comment (mseri) [inline packages/json-data-encoding/json-data-encoding.1.0.1/opam:12]:
```suggestion
  ["dune" "runtest" "-p" name "-j" jobs] {with-test & !ocaml-option-bytecode-only:installed}
```
For the exe error, this should be the problem

**Comment by mseri** [comment]
> what seems to be an actual issue on `s390x-ocaml-5.1`

Should we make it unavailable there?

**Comment by raphael-proust** [comment]
I've added the availability field.

I don't think it's an endianness issue because it works on s390x-ocaml.4.14. So it's probably a combination of multiple factors, endianness being only one of them?

**Comment by avsm** [comment]
Possibly https://github.com/ocaml/ocaml/pull/12919 for on s390x

**Comment by mseri** [comment]
What is the status? Can we merge this?

**Comment by raphael-proust** [comment]
@mseri I guess technically the package is fine but the s390x+ocaml.5.1 combination is broken. I'm ok merging this, but I'm also ok marking as `available: arch != "s390x" | ocaml-version != "5.1"`.

**Comment by mseri** [comment]
Please send an update to the availability field once a fixed ocaml is released

**Pull request merged**